src/framework/PreferenceStore/PreferenceStoreContext.tsx

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { useEffect } from 'react'
+import { useParams } from 'react-router-dom'
+
+import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
+import { useLocalStorage } from '@common/hooks'
+import type { UserInfo } from 'services/cd-ng'
+
+export interface PreferencePeripheralProps extends ProjectPathProps {
+  userId: string
+}
+
+export enum PreferenceScope {
+  USER = 'USER',
+  ACCOUNT = 'ACCOUNT',
+  ORG = 'ORG',
+  PROJECT = 'PROJECT',
+  MACHINE = 'MACHINE' // or workstation. This will act as default PreferenceScope
+}
+
+export interface PreferenceStoreOptions {
+  fromBackend?: boolean
+}
+
+export interface PreferenceStoreStateProps {
+  readonly currentUserInfo: UserInfo | undefined
+}
+
+/**
+ * Preference Store - helps to save ANY user-personalisation info
+ */
+export interface PreferenceStoreProps<T> {
+  set(scope: PreferenceScope, entityToPersist: string, value: T, options?: PreferenceStoreOptions): void
+  get(scope: PreferenceScope, entityToRetrieve: string, options?: PreferenceStoreOptions): T
+  clear(scope: PreferenceScope, entityToRetrieve: string, options?: PreferenceStoreOptions): void
+  updatePreferenceStore(data: PreferenceStoreStateProps): void
+}
+
+export interface ScopeContext {
+  accountId?: string
+  projectIdentifier?: string
+  orgIdentifier?: string
+  userId?: string
+}
+
+export interface PreferenceStoreContextProps<T> {
+  preference: T
+  setPreference: (value: T) => void
+  clearPreference: () => void
+  updatePreferenceStore: (data: PreferenceStoreStateProps) => void
+}
+
+export const PREFERENCES_TOP_LEVEL_KEY = 'preferences'
+
+export const PreferenceStoreContext = React.createContext<PreferenceStoreProps<any>>({
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232308)
```suggestion
export const PreferenceStoreContext = React.createContext<PreferenceStoreProps<unknown>>({
```